lib/matplotlib/widgets.py

```diff
@@ -574,38 +575,82 @@ def _clicked(self, event):
                 self.set_active(i)
                 break
 
-    def set_active(self, index):
+    def set_active(self, index, state=_TOGGLE):
         """
         Directly (de)activate a check button by index.
+            Default behaviour is to toggle the state,
+            which can be controlled with state=bool flag.
 
         *index* is an index into the original label list
             that this object was constructed with.
             Raises ValueError if *index* is invalid.
 
+        *state* is a boolean value to set the target state
+            of the button, regardless of its current state.
+            By default (no value specified),
+                the button state gets toggled (checked or not).
+            Only True or False values are allowed for *state*,
```

Comment:
Actually I was going to allow any value and just call bool() on it, but it's not a big deal.